lib/resolver.manifest.js

```diff
@@ -53,161 +56,157 @@ export default class PodletClientManifestResolver {
         return this.#metrics;
     }
 
-    resolve(outgoing) {
-        return new Promise(resolve => {
-            if (outgoing.status === 'cached') {
-                resolve(outgoing);
-                return;
+    async resolve(outgoing) {
+        if (outgoing.status === 'cached') {
+            return outgoing;
+        }
+
+        const headers = {
+            'User-Agent': UA_STRING,
+        };
+
+        const reqOptions = {
+            rejectUnauthorized: outgoing.rejectUnauthorized,
+            timeout: outgoing.timeout,
+            method: 'GET',
+            // agent: outgoing.manifestUri.startsWith('https://')
+            // ? this.#httpsAgent
+            // : this.#httpAgent,
+            json: true,
+            headers,
+        };
```

Comment:
Even though all tests are passing, I'm not super confident that there aren't small differences that will show through. Some of the reqOptions in places like this here will not be being respected by Undici

Reply:
I think that's completely fine at this point. In v6 we'll probably rework some parts of this anyway.